addons/Text_Selection/src/presenter.js

```diff
@@ -1719,33 +1735,148 @@ function AddonText_Selection_create() {
         presenter.printableState = JSON.parse(state);
     };
 
-    presenter.getPrintableHTML = function (model, showAnswers) {
-        var model = presenter.upgradeModel(model);
+    presenter.PRINTABLE_STATE_MODE = {
+        EMPTY: 0,
+        SHOW_ANSWERS: 1,
+        SHOW_USER_ANSWERS: 2,
+        CHECK_ANSWERS: 3
+    };
+
+   presenter.getPrintableHTML = function (model, showAnswers) {
+       var upgradedModel = presenter.upgradeModel(model);
+       var isActiveMarkPhrasesToSelectMode = isMarkPhrasesToSelectMode(upgradedModel);
```

Comment:
tutaj powinieneś raczej sparsować model za pomocą validateModel, zapisać sobie wynik tej metody (zwyczajowo zmienna jest nazywana configuration) i stamtąd wyciągać mode. W ten sposób będziesz miał pewność, że jeśli w przyszłości coś zmieni się w modelu, to będziesz miał aktualną wersję.

Reply:
Naprawione. Dzięki za uwagę. Faktycznie jest teraz lepiej.